units/patch_intent_test.go

```diff
@@ -328,22 +338,243 @@ func (s *PatchIntentUnitsSuite) TestGetPreviousPatchDefinition() {
 			DisplayTasks: []patch.DisplayTask{{Name: "dt2"}},
 		},
 	}}
-	vt, err := j.getPreviousPatchDefinition(&project, false)
+	currentPatchDoc := intent.NewPatch()
+	s.NoError(err)
+	previousPatchStatus, err := j.setToPreviousPatchDefinition(currentPatchDoc, &project, false)
+	s.NoError(err)
+	s.Equal(previousPatchStatus, "failed")
+
+	s.Equal(currentPatchDoc.BuildVariants, previousPatchDoc.BuildVariants)
+	s.Equal(currentPatchDoc.RegexBuildVariants, previousPatchDoc.RegexBuildVariants)
+	s.Equal(currentPatchDoc.RegexTasks, previousPatchDoc.RegexTasks)
+	s.Equal(currentPatchDoc.Tasks, previousPatchDoc.Tasks)
+
+	previousPatchStatus, err = j.setToPreviousPatchDefinition(currentPatchDoc, &project, true)
+	s.NoError(err)
+	s.Equal(previousPatchStatus, "failed")
+	s.Equal(currentPatchDoc.BuildVariants, previousPatchDoc.BuildVariants)
+	s.Equal(currentPatchDoc.RegexBuildVariants, previousPatchDoc.RegexBuildVariants)
+	s.Equal(currentPatchDoc.RegexTasks, previousPatchDoc.RegexTasks)
+	s.Equal(currentPatchDoc.Tasks, []string{"t1"})
+}
+
+func (s *PatchIntentUnitsSuite) TestBuildTasksandVariantsWithRepeat() {
```

Comment:
The "reuse" flag is also called the "repeat" flag, whereas "reuse-failed" is only called that, so I think this test should be RepeatFailed for clarity.